pkg/controllers/provisioningv2/rke2/provisioninglog/provisioninglog.go

```diff
@@ -44,17 +45,19 @@ type handler struct {
 }
 
 func (h *handler) OnConfigMap(key string, cm *corev1.ConfigMap) (*corev1.ConfigMap, error) {
-	if cm == nil {
+	if cm == nil || !cm.DeletionTimestamp.IsZero() {
 		return nil, nil
 	}
 	if cm.Name != provisioningLogName || (!clusterRegexp.MatchString(cm.Namespace) && cm.Namespace != "local") {
 		return cm, nil
 	}
 	provCluster, err := h.clusterCache.GetByIndex(clusterindex.ClusterV1ByClusterV3Reference, cm.Namespace)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return cm, err
+	}
 	if apierrors.IsNotFound(err) || len(provCluster) == 0 {
```

Comment:
To the reviewer: if err was not nil, `len(provCluster) == 0` was always true, so we never reenqueued ):